py-core/app/util/tezos_node_functions.py

```diff
@@ -0,0 +1,150 @@
+DOCKER_COMPOSE_FILE_PATH = "util/docker-compose/"
+
+# TODO: CHANGE THESE LINKS
+# Snapshot Download URLs
+MAINNET_FULL_SNAPSHOT = "https://conseil-snapshots.s3.amazonaws.com/tezos-node_snapshot-latest.full"
+MAINNET_ROLLING_SNAPSHOT = "https://conseil-snapshots.s3.amazonaws.com/tezos-node_snapshot-latest.full"
+CARTHAGENET_FULL_SNAPSHOT = "https://conseil-snapshots.s3.amazonaws.com/tezos-node_snapshot-latest.full"
+CARTHAGENET_ROLLING_SNAPSHOT = "https://conseil-snapshots.s3.amazonaws.com/tezos-node_snapshot-latest.full"
+
+# Data-Directory Download URLs
+MAINNET_DATA_DIR = "https://conseil-snapshots.s3.amazonaws.com/tezos-data.tar.gz"
+CARTHAGENET_DATA_DIR = "https://conseil-snapshots.s3.amazonaws.com/tezos-data.tar.gz"
+
+
+def create_node(data):
+
+    data_location = DOCKER_COMPOSE_FILE_PATH + data["name"]
+    filename = ""
+
+    # Start Node
+    if data["restore"]:
+        shutil.copytree(DOCKER_COMPOSE_FILE_PATH + "reference-snapshot", DOCKER_COMPOSE_FILE_PATH + data["name"])
+
+        if data["network"] == "mainnet" and data["history_mode"] == "archive":
+            filename = wget.download(MAINNET_DATA_DIR, data_location)
+
+        if data["network"] == "mainnet" and data["history_mode"] == "full":
+            filename = wget.download(MAINNET_FULL_SNAPSHOT, data_location)
+
+        if data["network"] == "mainnet" and data["history_mode"] == "rolling":
+            filename = wget.download(MAINNET_ROLLING_SNAPSHOT, data_location)
+
+        if data["network"] == "carthagenet" and data["history_mode"] == "archive":
+            filename = wget.download(CARTHAGENET_DATA_DIR, data_location)
+
+        if data["network"] == "carthagenet" and data["history_mode"] == "full":
+            filename = wget.download(CARTHAGENET_FULL_SNAPSHOT, data_location)
+
+        if data["network"] == "carthagenet" and data["history_mode"] == "rolling":
+            filename = wget.download(CARTHAGENET_ROLLING_SNAPSHOT, data_location)
+
+        if data["history_mode"] == "archive":
+            file = tarfile.open(data_location + "/tezos-data.tar.gz")
+            file.extractall(data_location)
+            file.close()
+            os.remove(data_location + "/tezos-data.tar.gz")
+        else:
+            docker_volumes = dict()
+
+            docker_volumes[os.getcwd() + DOCKER_COMPOSE_FILE_PATH + data["name"] + "/tezos-data"] = dict()
+            docker_volumes[os.getcwd() + DOCKER_COMPOSE_FILE_PATH + data["name"] + "/tezos-data"]["bind"] = "/var/run/tezos"
+            docker_volumes[os.getcwd() + DOCKER_COMPOSE_FILE_PATH + data["name"] + "/tezos-data"]["mode"] = "rw"
+
+            docker_volumes[os.getcwd() + DOCKER_COMPOSE_FILE_PATH + data["name"] + "/" + filename] = dict()
+            docker_volumes[os.getcwd() + DOCKER_COMPOSE_FILE_PATH + data["name"] + "/" + filename]["bind"] = "/snapshot"
+            docker_volumes[os.getcwd() + DOCKER_COMPOSE_FILE_PATH + data["name"] + "/" + filename]["mode"] = "rw"
```

Comment:
you can refactor `os.getwd() + ... "/"` to a single var and use that. Dry code ftw!